calyx/src/passes/dump_ports.rs

```diff
@@ -0,0 +1,52 @@
+use std::collections::HashMap;
+
+use crate::ir::{self, RRC, WRC};
+
+pub fn format_port_name(comp: &ir::Id, port: &ir::Id) -> ir::Id {
+    format!("{}_{}", comp.id, port.id).into()
+}
+
+pub fn dump_ports_to_signature(
+    component: &mut ir::Component,
+    f: fn(&RRC<ir::Cell>) -> bool,
+    port_names: &mut HashMap<ir::Id, HashMap<ir::Id, HashMap<ir::Id, ir::Id>>>,
+) {
+    let comp_name = component.name.clone();
+    let (ext_cells, cells): (Vec<_>, Vec<_>) =
+        component.cells.drain().partition(f);
+
+    component.cells.append(cells.into_iter());
+
+    for cell_ref in ext_cells {
+        let mut cell = cell_ref.borrow_mut();
+        let name = cell.name().clone();
+        let (ports_inline, _): (Vec<_>, Vec<_>) =
+            cell.ports.drain(..).partition(|p_ref| {
+                let p = p_ref.borrow();
+                p.attributes.get("clk").is_none()
```

Comment:
Document why we're skipping these ports. Link to the issue you created about this